component/remote/vault/vault.go

```diff
@@ -0,0 +1,309 @@
+package vault
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/pkg/flow/rivertypes"
+	"github.com/oklog/run"
+
+	vault "github.com/hashicorp/vault/api"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "remote.vault",
```

Comment:
> The authManager will need to re-authenticate to vault every once in a while so we can continue getting the newest value stored in the secret.

Yes. Authentication tokens (also a `*vault.Secret`) can have leases which are used to renew them. My understanding is that authenticating against Vault can have side effects depending on the authentication method you're using, so it's better to renew your authentication lease whenever possible instead of always re-authenticating whenever it's time to reread the secret.